### PR TITLE
Add type definitions for Steps.Step

### LINKED ISSUE #14442: Add type definitions for Steps.Step
- [X] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently, Steps.Step is defined as `static Step: any`, as present in antd/lib/steps/index.d.ts

The type definitions should be updated to conform with the [documentation](https://ant.design/components/steps/)

### What does the proposed API look like?
```
static Step: {
    description: string | React.ReactNode;
    icon: string | React.ReactNode;
    status: string;
    title: string | React.ReactNode;
};
```

### PR DESCRIPTION
**Pull request by wx1322**
### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

Follow https://github.com/ant-design/ant-design/issues/14442

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/steps/index.tsx [hunk 1]
```diff
@@ -19,7 +19,12 @@ export interface StepsProps {
 }
 
 export default class Steps extends React.Component<StepsProps, any> {
-  static Step = RcSteps.Step;
+  static Step = {
+    description: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    status: PropTypes.string,
+  };
```
Comment (zombieJ) [inline components/steps/index.tsx]: `RcSteps.Step` is a component, not a propTypes. You need define an typescript def like:
```tsx
static Step = RcSteps.Step as React.ClassicComponentClass<StepProps>;
```

Comment (wx1322) [reply components/steps/index.tsx]: ok, let me try!

File: components/steps/index.tsx [hunk 2]
```diff
@@ -18,8 +18,15 @@ export interface StepsProps {
   style?: React.CSSProperties;
 }
 
+export interface StepProps {
+  description?: string | React.ReactNode;
```
Comment (afc163) [inline components/steps/index.tsx]: `React.ReactNode` includes `string`.

Comment (wx1322) [reply components/steps/index.tsx]: Thanks for your suggestion!

File: components/steps/index.tsx [hunk 3]
```diff
@@ -18,8 +18,15 @@ export interface StepsProps {
   style?: React.CSSProperties;
 }
 
+export interface StepProps {
+  description?: React.ReactNode;
+  icon?: React.ReactNode;
+  status?: string;
```
Comment (anton62k) [inline components/steps/index.tsx]: Field status should be "wait", "process", "finish" or "error".
https://ant.design/components/steps/#Steps.Step

Comment (wx1322) [reply components/steps/index.tsx]: I will fix it!

**Review by kashirov — APPROVED** [review]

**Pull request merged**